software/util/reorg.py

```diff
@@ -6,67 +6,104 @@
 """
 
 from collections.abc import Sequence
-from rdflib import Graph, Namespace
+from rdflib import Graph, Namespace, RDF, RDFS, DCTERMS, URIRef
+import argparse
+import itertools
 import logging
 import sys
-import argparse
 
-def SchemaOrgGraph():
-  """Creates a graph with usual namespaces set as we want in schemaorg."""
-  g = Graph()
-  g.bind("dc", Namespace("http://purl.org/dc/terms/"), replace=True)
-  return g
+class SchemaOrgGraph(object):
+  def __init__(self, filename = None, format = "turtle"):
+    self.g = Graph()
+    # Binding it here, as by default it would bind the /elements/1.1/ instead
+    # of the dc terms. this way, the elements get assigned 'dc1' or such
+    # as a prefix, and we do not use that.
+    self.g.bind("dc", Namespace("http://purl.org/dc/terms/"), replace=True)
+    if filename:
+      self.g.parse(filename, format=format)
+
+  def __getattr__(self, *args, **kwargs):
+    return getattr(self.g, *args, **kwargs);
+
+  def IdenticalTo(self, other):
+    only_in_other = other.g - self.g
+    only_in_self = self.g - other.g
+    if len(only_in_other) or len(only_in_self):
+      raise f"Graphs differ: {only_in_other + only_in_self}"
+    return True
+
+  def FullyContains(self, graph):
```

Comment:
Weird function signature, either returns True or an exception. Do  we need the True?

Reply:
The goal is to use it in constructs like:

```
if my_graph.FullyContains(other_graph):
  # do something
```

and still get a sensible error message when that fails.